### ISSUE TITLE
ROUND_ROBIN fallback not working

### ISSUE THREAD

**Issue by ajw227** [arangodb/arangodb-java-driver#525]
### Description

In case of coordinator is down, the fallback logic does not work properly.

1. getDocument() : work properly. (GET)
2. query() : Error. (POST)

Looking at the code, it seems that the POST method is excluded. Please review.

#### Java driver

```
    private final ArangoDB arangoDB = new ArangoDB.Builder()
        .host("localhost", 8529)
        .host("localhost", 8530)
        .host("localhost", 8531) //Invalid host
        .loadBalancingStrategy(LoadBalancingStrategy.ROUND_ROBIN)
        .build();
```

### Test code

```
    private String getQueryValue() {
        var cursor = arangoDB.db().query("RETURN \"value\"", String.class);
        var list =  cursor.asListRemaining();
        try {
            cursor.close();
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
        return list.get(0);
    }

    public void test() {
        var first = getQueryValue(); // OK.
        var second = getQueryValue(); // OK.
        var third = getQueryValue(); //Connection refused: localhost/127.0.0.1:8531
    }
```

### Related code

https://github.com/arangodb/arangodb-java-driver/blob/main/http/src/main/java/com/arangodb/http/HttpCommunication.java#L171

```
    private boolean isSafe(final InternalRequest request) {
        RequestType type = request.getRequestType();
        return type == RequestType.GET || type == RequestType.HEAD || type == RequestType.OPTIONS;
    }
```

**Comment by ajw227** [comment]
@rashtao Thx for review.
I found the issue.

I tested  **7.1.0**.
Changed connect exception handling.

https://github.com/arangodb/arangodb-java-driver/commit/94a7d4e928d2f9573d4e93464dff82aea67d8678.

After changed to 7.2.0, everything is ok. 👍

**Comment by rashtao** [comment]
Happy to help, but note that these changes have been released in 7.3.0.
